lua/treesitter-context.lua

```diff
@@ -153,9 +191,18 @@ function M.enable()
     end
   end)
 
-  autocmd({ 'BufLeave', 'WinLeave' }, close)
+  autocmd({ 'WinClosed' }, function(args)
+    local winid = tonumber(args.match)
+    for stored_winid, _ in pairs(require('treesitter-context.render').get_window_contexts()) do
+      if winid == stored_winid then
+        close(stored_winid)
+      end
+    end
+  end)
```

Comment:
Could we keep the same pattern as other events, and assign the function to a name and pass a ref to it here, instead of defining it literally here? Keeps things more ordered and helps reify what's happening.